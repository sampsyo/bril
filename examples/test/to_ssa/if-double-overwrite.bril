@main(foo: bool) {
.entry:
    cond: bool = const true;
    b: int = const 42;
    br cond .true .false;
.true:
    b: int = const 1;
    br foo .left .right;
.false:
    b: int = const 0;
    jmp .zexit;
.left:
    b: int = const 2;
    jmp .zexit;
.right:
    b: int = const 3;
    jmp .zexit;
.zexit:
    print b;
}
