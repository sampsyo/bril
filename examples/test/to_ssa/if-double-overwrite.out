@main(foo: bool) {
.entry:
  cond.0: bool = const true;
  b.0: int = const 42;
  br cond.0 .true .false;
.true:
  b.2: int = const 1;
  br foo .left .right;
.false:
  b.1: int = const 0;
  jmp .zexit;
.left:
  b.3: int = const 2;
  jmp .zexit;
.right:
  b.4: int = const 3;
  jmp .zexit;
.zexit:
  b.5: int = phi b.1 b.3 b.4 .false .left .right;
  print b.5;
  ret;
}
